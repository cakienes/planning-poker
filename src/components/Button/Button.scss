.defaultButtonHolder {
    margin-top: 40px;
    button {
        outline: none;
        width: 200px;
        font-size: 24px;
        color: #000;
        padding: 10px;
        background-color: #fff;
    }
}
