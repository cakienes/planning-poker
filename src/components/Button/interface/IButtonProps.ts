export default interface IButtonProps {
    label?: string;
    onClick?: () => void;
    type: 'submit' | 'reset' | 'button';
}
