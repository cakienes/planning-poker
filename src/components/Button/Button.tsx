import React from 'react';
import './Button.scss';
import IButtonProps from './interface/IButtonProps';

const Button = (props: IButtonProps) => {
    return (
        <div className="defaultButtonHolder">
            <button onClick={props.onClick}>{props.label}</button>
        </div>
    );
};

export default Button;
