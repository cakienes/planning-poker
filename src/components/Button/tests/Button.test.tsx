import { shallow } from 'enzyme';
import React from 'react';
import Button from '../Button';
import IButtonProps from '../interface/IButtonProps';

describe('Button.tsx', () => {
    const onClickMock = jest.fn();
    const buttonProps: IButtonProps = {
        label: 'label',
        onClick: onClickMock,
        type: 'button',
    };

    it('renders without crashing', async () => {
        expect(shallow(<Button {...buttonProps} />)).toMatchSnapshot();
    });
});
