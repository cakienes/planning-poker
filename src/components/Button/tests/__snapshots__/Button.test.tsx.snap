// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx renders without crashing 1`] = `ShallowWrapper {}`;
